### DISCUSSION TITLE
Nuxt 3 RC.12 & octokit problem

### DISCUSSION THREAD

**Discussion by gaminglifex** [nuxt/nuxt#16522] [Questions]
### Environment

- Operating System: `Windows_NT`
- Node Version:     `v16.16.0`
- Nuxt Version:     `3.0.0-rc.12`
- Nitro Version:    `0.6.0`
- Package Manager:  `yarn@1.22.18`
- Builder:          `vite`
- User Config:      `modules`
- Runtime Modules:  `@nuxtjs/tailwindcss@6.0.1`
- Build Modules:    `-`

### Reproduction

https://stackblitz.com/edit/nuxt-starter-4as5ky?file=pages/index.vue

### Describe the bug

Trying to implement octokit.js into Nuxt 3, no ideas whether it is a bug or I did it wrongly.

### Additional context

![image](https://user-images.githubusercontent.com/72642816/198475089-88f75b3f-8e53-4738-be75-8f76fec0fe0e.png)

**Comment by danielroe** [comment]
This looks like an upstream issue to me. Among other issues:

1. the octokit family of libraries seems to assume that anything using `module` (ESM) is web, but that anything using `main` (CJS) is node-based. [octokit](https://unpkg.com/browse/octokit@2.0.10/package.json)
2. some of the 'web' dependencies seem to include node deps (this is what the errors are about in relation to stream are)

**Reply by gaminglifex** [reply]
Thank you for your reply. I found some body on github is able to implement octokit into their nuxt3. May I know how? https://github.com/Tch1b0/JP-nuxt

**Reply by TheAlexLichter** [reply]
Just FYI, I also use OctoKit with Nuxt 3 - adding it to several of Nitro's endpoints.

**Reply by gaminglifex** [reply]
Thank you! I am totally a newbie to webdev. 
I am sorry that I still do not have a clear direction of what I should do to make it works. May I have some resources/references on how to add ocotkit to Nitro's endpoints?

**Reply by gaminglifex** [reply]
Thank you all I think I am able to make it, at least able to fetch some information from a private repo without any error message by adding apis to "server".

**Reply by toniengelhardt** [reply]
> Just FYI, I also use OctoKit with Nuxt 3 - adding it to several of Nitro's endpoints.

Hi @manniL, is one of them by chance open-source to peak in?

**Reply by TheAlexLichter** [reply]
@toniengelhardt No, *but* feel free to have a look in https://github.com/antfu/issue-up/blob/main/api/webhook.post.ts

**Reply by toniengelhardt** [reply]
> @toniengelhardt No, _but_ feel free to have a look in https://github.com/antfu/issue-up/blob/main/api/webhook.post.ts

Thanks! 

I do basically the same as in the repo you sent, just a bit simpler. I have a composable like

```ts
import { graphql } from '@octokit/graphql'

export const useOctoGqlQ = async (query: string) => {
  const cookieValue = useGithubCookie().value
  return graphql(query, { headers: { authorization: `token ${cookieValue}` } })
}
```

but I get `'global' is not defined`. 

Same with the Octokit SDK.

The repo you posted is not a Nuxt app though, maybe the issue is there.
